bonsai-core/src/main/scala/com/stripe/bonsai/FullBinaryTree.scala

```diff
@@ -6,36 +6,107 @@ import scala.collection.immutable.Queue
 
 import com.stripe.bonsai.layout.Vec
 
+/**
+ * This type represents a binary tree, where each branch is guaranteed
+ * to have two children (i.e. it is "full").
+ *
+ * Members:
+ *
+ *  - bitset: encodes the tree structure using a concept of internal
+ *    and external nodes, e.g. a (potentially unbalanced) heap.  it's
+ *    sort of complicated. basically, for non-balanced trees, we need
+ *    to add "spacing" so that the heap addressing invariants are
+ *    preserved.
+ *
+ *  - isLeaf: encodes whether an (internal) node is a leaf or branch
+ *    node.
+ *
+ *  - branchLabels: a dense array of branch label values to index
+ *    into. we use isLeaf to convert 'bitset indices' into
+ *    'branchLabels' indices.
+ *
+ *  - leafLabels: a dense array of leaf label values to index into. we
+ *    use isLeaf to convert 'bitset indices' into 'leafLabels' indices.
+ *
+ * Example 1:
+ *
+ *       a
```

Comment:
Also, probably worth it to wrap these in `{{{ ... }}}`, so we get monospaced font.